app/grandchallenge/components/forms.py

```diff
@@ -79,3 +86,156 @@ def save(self, *args, **kwargs):
 
     class Meta:
         fields = ("user_upload", "creator", "comment")
+
+
+class MultipleCIVCreateForm(Form):
+    _possible_widgets = {
+        *InterfaceFormField._possible_widgets,
+    }
+
+    def __init__(self, *args, instance, base_obj, user, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.instance = instance
+        self.user = user
+        self.base_obj = base_obj
+
+        # add fields for all interfaces that already exist on
+        # other display sets / archive items
+        for slug, values in base_obj.values_for_interfaces.items():
+            current_value = None
+
+            if instance:
+                current_value = instance.values.filter(
+                    interface__slug=slug
+                ).first()
+
+            self.init_interface_field(
+                interface_slug=slug, current_value=current_value, values=values
+            )
+        # Add fields for dynamically added new interfaces:
+        # These are sent along as form data like all other fields, so we can't
+        # tell them apart from the form fields initialized above. Hence
+        # the check if they already have a corresponding field on the form or not.
+        for slug in self.data.keys():
+            if (
+                ComponentInterface.objects.filter(slug=slug).exists()
+                and slug not in self.fields.keys()
+            ):
+                self.init_interface_field(
+                    interface_slug=slug,
+                    current_value=None,
+                    values=[],
+                )
+
+    def save(self):
+        # we need a dummy save method for form_valid() when called from a
+        # generic CreateView/UpdateView
+        pass
```

Comment:
Is this still necessary now that you switched to a `TemplateView`?

Reply:
Yes, because this is the form for the DisplaySetCreate and DisplaySetUpdate views.

Reply:
> Doesn't it mean that you skip saving if you do add this mixin to a form that actually updates something? Again, you seem to be fighting the framework too much with this design.

That is true. Maybe changing the Create and Update views to FormViews would be more appropriate then. It would not require the dummy save method on the form mixin, and in the end neither the create nor the update views for display sets are traditional create and update views, since the form is dynamically constructed.

Reply:
I turned the create and update views into FormViews and have also updated the form names. Hope this clarifies things.